### PR TITLE
Fortinet's new module for fortios_system_ha

### PR DESCRIPTION
**Pull request by migumun**
##### SUMMARY
 
Fortinet is adding Ansible support for FortiOS and FortiGate products. This module follows the same structure, guidelines and ideas given in previous approved module for a parallel feature of FortiGate (user device): https://github.com/ansible/ansible/pull/56447
In this case we are providing a different functionality: "Fortios System HA".
 
Please note that this will be part of other modules to come for FortiGate, including different functionalities: system, wireless-controller, firewall, webfilter, ips, web-proxy, wanopt, application, dlp spamfilter, log, vpn, certificate, user, dnsfilter, antivirus, report, waf, authentication, switch controller, endpoint-control and router. We plan to follow the same style, structure and usage as in the previous module in order to make it easier to comply with Ansible guidelines.
 
This PR has been updated according to suggestions and code reviews given in previous submissions of Fortinet modules.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

fortios_system_ha

##### ANSIBLE VERSION

```
ansible 2.8.2
python version = 3.7.3 (default, Apr  3 2019, 05:39:12) [GCC 8.3.0]
```

### CODE REVIEW

**Comment by thomnico** [comment]
shipit

**Review by NilashishC — CHANGES_REQUESTED** [review]
`state` key seems to be missing.

File: lib/ansible/modules/network/fortios/fortios_system_ha.py [hunk 1]
```diff
@@ -0,0 +1,902 @@
...
+                description:
+                    - Normally you would only reduce this value for failover testing.
+                type: int
+            hb_interval:
+                description:
+                    - Time between sending heartbeat packets (1 _ 20 (100*ms)). Increase to reduce false positives.
+                type: int
+            hb_lost_threshold:
+                description:
+                    - Number of lost heartbeats to signal a failure (1 _ 60). Increase to reduce false positives.
```
Comment (NilashishC) [inline lib/ansible/modules/network/fortios/fortios_system_ha.py]: probably?
```suggestion
                    - Number of lost heartbeats to signal a failure (1 - 60). Increase to reduce false positives.
```

File: lib/ansible/modules/network/fortios/fortios_system_ha.py [hunk 2]
```diff
@@ -0,0 +1,902 @@
...
+                description:
+                    - Heartbeat interfaces. Must be the same for all members.
+                type: str
+            hc_eth_type:
+                description:
+                    - Transparent mode HA heartbeat packet Ethertype (4_digit hex).
+                type: str
+            hello_holddown:
+                description:
+                    - Time to wait before changing from hello to work state (5 _ 300 sec).
```
Comment (NilashishC) [inline lib/ansible/modules/network/fortios/fortios_system_ha.py]: probably?
```suggestion
                    - Time to wait before changing from hello to work state (5 - 300 sec).
```

File: lib/ansible/modules/network/fortios/fortios_system_ha.py [hunk 3]
```diff
@@ -0,0 +1,902 @@
...
+                choices:
+                    - enable
+                    - disable
+            ha_uptime_diff_margin:
+                description:
+                    - Normally you would only reduce this value for failover testing.
+                type: int
+            hb_interval:
+                description:
+                    - Time between sending heartbeat packets (1 _ 20 (100*ms)). Increase to reduce false positives.
```
Comment (NilashishC) [inline lib/ansible/modules/network/fortios/fortios_system_ha.py]:
```suggestion
                    - Time between sending heartbeat packets (1 - 20 (100*ms)). Increase to reduce false positives.
```

File: lib/ansible/modules/network/fortios/fortios_system_ha.py [hunk 4]
```diff
@@ -0,0 +1,902 @@
...
+            gratuitous_arps:
+                description:
+                    - Enable/disable gratuitous ARPs. Disable if link_failed_signal enabled.
+                type: str
+                choices:
+                    - enable
+                    - disable
+            group_id:
+                description:
+                    - Cluster group ID  (0 _ 255). Must be the same for all members.
```
Comment (NilashishC) [inline lib/ansible/modules/network/fortios/fortios_system_ha.py]:
```suggestion
                    - Cluster group ID  (0 - 255). Must be the same for all members.
```

File: lib/ansible/modules/network/fortios/fortios_system_ha.py [hunk 5]
```diff
@@ -0,0 +1,902 @@
...
+        default: true
+    system_ha:
+        description:
+            - Configure HA.
+        default: null
+        type: dict
+        suboptions:
+            arps:
+                description:
+                    - Number of gratuitous ARPs (1 _ 60). Lower to reduce traffic. Higher to reduce failover time.
```
Comment (NilashishC) [inline lib/ansible/modules/network/fortios/fortios_system_ha.py]:
```suggestion
                    - Number of gratuitous ARPs (1 -60). Lower to reduce traffic. Higher to reduce failover time.
```

File: lib/ansible/modules/network/fortios/fortios_system_ha.py [hunk 6]
```diff
@@ -0,0 +1,902 @@
...
+                type: int
+            arps_interval:
+                description:
+                    - Time between gratuitous ARPs  (1 _ 20 sec). Lower to reduce failover time. Higher to reduce traffic.
```
Comment (NilashishC) [inline lib/ansible/modules/network/fortios/fortios_system_ha.py]:
```suggestion
                    - Time between gratuitous ARPs  (1 - 20 sec). Lower to reduce failover time. Higher to reduce traffic.
```

**Review by NilashishC — COMMENTED** [review]
The ranges in the documentation seem to have `_` instead of `-`. Could you please have a look?

**Pull request closed without merging**